### PR TITLE
fix(sql): change default import work dir location to avoid extra file copying

### PR DESCRIPTION
**Pull request by puzpuzpuz**
The import work directory defaulted to the import root directory (the one where CSV files are located), unless specified explicitly. This default was problematic in scenarios when the import root directory and database's own root directory were located on different disks (or different file systems). In such case, moving partition files from the import work directory to the database directory requires expensive file copying instead of a cheap rename operation.

To avoid the problem, this PR changes the default value for the `cairo.sql.copy.work.root` property to `<root_dir>/tmp`.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 13 / 15 (86.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|10|11|90.91%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 15 (93.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 15 (93.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 15 (93.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|11|11|100.00%|

File: core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java [hunk 1]
```diff
@@ -603,12 +603,21 @@ private void collectStub(final TextImportTask task) {
...
+            if (errno != 0) {
+                throw CairoException.critical(errno).put("could not create import work root directory [path='").put(workDirPath).put("', errno=").put(errno).put("]");
+            }
+        }
 
-        Path workDirPath = tmpPath.of(importRoot).slash$();
+        // Next, remove and recreate the per-table sub-dir.
+        removeWorkDir();
+        workDirPath = tmpPath.of(importRoot).slash$();
         int errno = ff.mkdir(workDirPath, configuration.getMkDirMode());
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java:618]: mkdir() does not return `errno`. This is a status code, 0 success, everything else error. `errno` is ff.errno()

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java:618]: Good stop. Let me fix it in the original and new code.

Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cutlass/text/ParallelCsvFileImporter.java]: Done in 052d88ba088fd535b10c17abdfb887d8d6eef234

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|13|13|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**